src/timers.c

```diff
@@ -1,32 +1,30 @@
 #define _GNU_SOURCE
 #include "timers.h"
 #include <stdio.h>
-#include <sys/epoll.h>
-#include <sys/timerfd.h>
 #include <time.h>
 #include <stdbool.h>
 #include <unistd.h>
+#include "uvbloop.h"
 
 
 #define MAXEVENTS 64
 
 
 void *timer_loop(void *ptr) {
-    struct epoll_event *events;
-    if((events = calloc(MAXEVENTS, sizeof(struct epoll_event))) == NULL) {
+    uvbloop_event_t *events;
+    if((events = calloc(MAXEVENTS, sizeof(uvbloop_event_t))) == NULL) {
         perror("calloc");
         return NULL;
     }
     timer_mgr_t *p = (timer_mgr_t *)ptr;
     int waiting;
     timer_entry_t *entry = NULL;
-    uint64_t exp;
     while(true) {
-        waiting = epoll_wait(p->epoll_fd, events, MAXEVENTS, -1);
+        waiting = uvbloop_wait(p->loop, events, MAXEVENTS);
         if(waiting > 0) {
             for(int i=0; i<waiting; i++) {
-                entry = (timer_entry_t *)events[i].data.ptr;
-                read(entry->tfd, &exp, sizeof(uint64_t));
+                entry = (timer_entry_t *)uvbloop_event_data(events);
```

Comment:
Or rather, this being the timer thread, uvbloop_wait nevents could probably just be 1